crates/broker/src/lib.rs

```diff
@@ -462,7 +472,7 @@ where
             chain_monitor.clone(),
         ));
         supervisor_tasks.spawn(async move {
-            task::supervisor(1, order_picker).await.context("Failed to start order picker")?;
+            Supervisor::new(order_picker).spawn().await.context("Failed to start order picker")?;
```

Comment:
Curious the thinking behind which services are critical vs not. I mean isn't every service critical for generating proofs except the off-chain market monitor?

Reply:
Critical might not be the best naming, but I added those to the ones I did because:
- It affects already locked stake, not as urgent for sth like order picker because especially on a failure there isn't a rush to get the service up again asap
- Those "critical" ones are cheap to startup, and don't have external calls to make to initialize, so there isn't really a reason to wait too long before retrying (keeping as close to immediately restarting as it was before)